lib/src/socket_connector.dart

```diff
@@ -97,6 +98,9 @@ class SocketConnector {
     if (closed) {
       throw StateError('Connector is closed');
     }
+    unawaited(thisSide.socket.done
+        .then((v) => _closeSide(thisSide))
+        .catchError((err) => _closeSide(thisSide)));
```

Comment:
As soon as we get reference to the socket, ensure there will be no uncaught exceptions